modules/base/rendering/renderablemodel.cpp

```diff
@@ -279,103 +282,173 @@ RenderableModel::RenderableModel(const ghoul::Dictionary& dictionary)
         }
     }
 
-    if (dictionary.hasKey(KeyGeomModelFile)) {
-        std::string file;
+    // Import Model from file
+    std::string file = absPath(p.geometryFile.string());
+    _geometry = ghoul::io::ModelReader::ref().loadModel(
+        file,
+        ghoul::io::ModelReader::ForceRenderInvisible(_forceRenderInvisible),
+        ghoul::io::ModelReader::NotifyInvisibleDropped(_notifyInvisibleDropped)
+    );
 
-        if (dictionary.hasValue<std::string>(KeyGeomModelFile)) {
-            // Handle single file
-            file = absPath(dictionary.value<std::string>(KeyGeomModelFile));
-            _geometry = ghoul::io::ModelReader::ref().loadModel(
-                file,
-                ghoul::io::ModelReader::ForceRenderInvisible(_forceRenderInvisible),
-                ghoul::io::ModelReader::NotifyInvisibleDropped(_notifyInvisibleDropped)
-            );
+    if (p.modelScale.has_value()) {
+        if (std::holds_alternative<Parameters::ScaleUnit>(*p.modelScale)) {
+            Parameters::ScaleUnit scaleUnit =
+                std::get<Parameters::ScaleUnit>(*p.modelScale);
+
+            switch (scaleUnit) {
+                case Parameters::ScaleUnit::Nanometer:
+                    _modelScale = DistanceUnit::Nanometer;
+                    break;
+                case Parameters::ScaleUnit::Micrometer:
+                    _modelScale = DistanceUnit::Micrometer;
+                    break;
+                case Parameters::ScaleUnit::Millimeter:
+                    _modelScale = DistanceUnit::Millimeter;
+                    break;
+                case Parameters::ScaleUnit::Centimeter:
+                    _modelScale = DistanceUnit::Centimeter;
+                    break;
+                case Parameters::ScaleUnit::Decimeter:
+                    _modelScale = DistanceUnit::Decimeter;
+                    break;
+                case Parameters::ScaleUnit::Meter:
+                    _modelScale = DistanceUnit::Meter;
+                    break;
+                case Parameters::ScaleUnit::Kilometer:
+                    _modelScale = DistanceUnit::Kilometer;
+                    break;
+                default:
+                    throw ghoul::MissingCaseException();
+            }
+            _scaleVector = glm::dvec3(convertUnit(_modelScale, DistanceUnit::Meter));
         }
-        else if (dictionary.hasValue<ghoul::Dictionary>(KeyGeomModelFile)) {
-            LWARNING("Loading a model with several files is deprecated and will be "
-                "removed in a future release"
-            );
+        else if (std::holds_alternative<float>(*p.modelScale)) {
+            _scaleVector = glm::dvec3(std::get<float>(*p.modelScale));
+        }
+    }
```

Comment:
I started to put a 
```cpp
else {
  throw ghoul::MissingCaseException();
}
```
in here as well so that there will be a hard crash when someone adds something to the variant but forgets to check it here